### PR TITLE
KNNClassifier

### PR DESCRIPTION
**Pull request by tedmoore**

### CODE REVIEW

File: src/routes/reference/knnclassifier.svx [hunk 1]
```diff
@@ -9,4 +9,30 @@ flair: reference
 category: Analyse Data
 ---
 
+<script>
+    import IndentNote from '$lib/components/IndentNote.svelte';
+</script>
 
+KNNClassifier is a supervised machine learning algorithm for classifying data points to learned categories. It uses an internal [KDTree](/reference/kdtree) to find the _k_ nearest neighbours of a point that needs classification (where _k_ is an integer >= 1). Whichever category, or "class", is most common among the neighbours is predicted as the category for that point. If an even number of `numNeighbours` is requested and there is a tie, the label with the closer point will be predicted. The parameter `weight` indicates whether or not the prediction should be weighted by the neighbours' distances.
```
Comment (jamesb93) [inline src/routes/reference/knnclassifier.svx:16]: k nearest should be k-nearest

File: src/routes/reference/knnclassifier.svx [hunk 2]
```diff
@@ -9,4 +9,30 @@ flair: reference
...
+Whenever training a machine learning model using supervised learning, it may be a good idea to create a [Training-Testing Split](/learn/training-testing-split) of the data.
+
+</IndentNote>
+
+## KNNClassifier vs. MLPClassifier
+
+FluCoMa includes another object for classification, the [MLPClassifier](/reference/mlpclassifier), which also uses supervised learning for classification. The KNN object works quite differently from the MLP object, each having their strengths and weaknesses. The main differences to know are that:
+
+1. the flexibility of the MLP objects make them generally more capable of learning complex relationships between inputs and outputs,
+2. the MLP objects involve more parameters and will take much longer to `fit` (aka. train) than the KNN objects, and
```
Comment (jamesb93) [inline src/routes/reference/knnclassifier.svx:37]: Would remove, and

**Comment by tedmoore** [comment]
> We could loop around in summer to make a visualisation of how this one works.

sounds good. while making this i imagined it but didn't go to implement. I think it could be a not to complicated extension of the kdtree gui.

**Comment by jamesb93** [comment]
> > We could loop around in summer to make a visualisation of how this one works.
> 
> sounds good. while making this i imagined it but didn't go to implement. I think it could be a not to complicated extension of the kdtree gui.

That's what I imagined as well. I'll make an issue and we can loop when we are less pressed?

**Pull request merged**